src/groups/mqb/mqba/mqba_clientsession.cpp

```diff
@@ -2727,7 +2731,55 @@ void ClientSession::processEvent(const bmqp::Event& event,
 {
     // executed by the *IO* thread
 
-    if (event.isControlEvent()) {
+    if (!event.isAuthenticationEvent() && !d_authenticationContext) {
```

Comment:
Can this ever happen?  We've gotten to a client session, and we're not authenticated?  I thought it was an invariant of our design that this cannot happen.

If so, we shouldn't log an error and ignore the message.